tremor-pipeline/src/op/trickle/select.rs

```diff
@@ -646,221 +648,647 @@ impl Operator for TrickleSelect {
                                     group: group_value.clone_static(),
-                                    id: idgen.next_id(),
+                                    id: idgen.next_id(), // after all this is a new event
                                 }),
                             )
                         });
-                    for (aggr, this) in prev_group.aggrs.iter().zip(this_group.aggrs.iter_mut()) {
-                        this.invocable.merge(&aggr.invocable).map_err(|e| {
+                    let window_event = this_group.window.on_event(&event)?;
+                    if window_event.emit && !window_event.include {
+                        // push
+                        let env = Env {
+                            context: &ctx,
+                            consts: &consts,
+                            aggrs: &this_group.aggrs,
+                            meta: &node_meta,
+                            recursion_limit: tremor_script::recursion_limit(),
+                        };
+                        let result = stmt.target.run(
+                            opts,
+                            &env,
+                            unwind_event,
+                            state,
+                            event_meta,
+                            &local_stack,
+                        )?;
+
+                        if let Some(guard) = &stmt.maybe_having {
+                            let test =
+                                guard.run(opts, &env, &result, state, &NULL, &local_stack)?;
+                            if let Some(test) = test.as_bool() {
+                                if !test {
+                                    continue;
+                                }
+                            } else {
+                                let s: &Select = &stmt;
+                                return tremor_script::errors::query_guard_not_bool(
+                                    s, guard, &test, &node_meta,
+                                )?;
+                            }
+                        }
+                        let result = result.into_owned();
+                        events.push((
+                            OUT,
+                            Event {
+                                id: this_group.id.clone(),
+                                ingest_ns: event.ingest_ns,
+                                // TODO: this will blindly chose the origin uri of the last event arriving at the window, which is at least misleading and might be just wrong
+                                // e.g. in case of having two source input events into a stream we select from
+                                // having a single origin_uri doesnt make sense here anymore, if this info is valuable, we should track multiple origins for such events
+                                //origin_uri: event.origin_uri.clone(),
+                                origin_uri: None,
+                                is_batch: event.is_batch,
```

Comment:
I think this might be off, since it's not strictly batched by a batching function and thus might not be able to be unbatched, we should set this to false

Reply:
yep, we should consider this a totally new event. Shake off your heritage, little event. be free! go brrrr!